### PR TITLE
ZEP-0008 Introduce `zarf dev inspect` and change `zarf package inspect` to a parent command

### PR DESCRIPTION
**Pull request by AustinAbro321**
- One-line PR description: Introduce ZEP to change `zarf package inspect` to a parent command and add command `zarf dev inspect`

- Issue link: #8

### CODE REVIEW

File: 8-show-manifest-and-values/README.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+```
+Usage:
+  zarf dev show-manifests [ DIRECTORY ] [flags]
+
+Flags:
+      --create-set stringToString   Specify package variables to set on the command line (KEY=value) (default [])
+      --deploy-set stringToString   Specify deployment variables to set on the command line (KEY=value) (default [])
+  -f, --flavor string               The flavor of components to include in the resulting package (i.e. have a matching or empty "only.flavor" key)
+      --kube-version                Override the default helm template KubeVersion when performing a package chart template
+      --confirm                     Confirms command without prompting. Skips prompts to configure variables.
```
Comment (soltysh) [inline 8-show-manifest-and-values/README.md:170]: Again, `--confirm` in a preview seems like an overkill.

File: 8-show-manifest-and-values/README.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+Below is the intended help text for `zarf package show-manifests`. `zarf package show-values-files` will include the same flags.
+```
+Usage:
+  zarf package show-manifests [ PACKAGE_SOURCE ] [flags]
+
+Flags:
+      --set stringToString          Specify deployment variables to set on the command line (KEY=value) (default [])
+      --kube-version                Override the default helm template KubeVersion when performing a package chart template
+      --components                  Comma-separated list of components whose manifests should be displayed.  Adding this flag will skip the prompts for selected components.  Globbing component names with '*' and deselecting 'default' components with a leading '-' are also supported
+      --confirm                     Confirms command without prompting. Skips prompts to configure variables and select optional components
```
Comment (soltysh) [inline 8-show-manifest-and-values/README.md:182]: Ditto.

File: 8-show-manifest-and-values/README.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+<!--
+Detail the things that people will be able to do if this ZEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a creator of Zarf packages, I want to make sure the variables in my package can get templated properly for the expected values of the deployers. I want to check this for both manifests and values files so I run `zarf dev show-manifests path/to/package-dir --deploy-set=MY_VAR=my-val` and `zarf dev show-values-files path/to/package-dir --deploy-set=MY_VAR=my-val`
```
Comment (soltysh) [inline 8-show-manifest-and-values/README.md:196]:
```suggestion
As a creator of Zarf packages, I want to make sure the variables in my package are properly rendered with the expected values. I want to check this for both manifests and values files so I run `zarf dev show-manifests path/to/package-dir --deploy-set=MY_VAR=my-val` and `zarf dev show-values-files path/to/package-dir --deploy-set=MY_VAR=my-val`
```

File: 8-show-manifest-and-values/README.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+
+#### Story 2
+
+As a deployer of Zarf packages, I want to check that the variables I intend to deploy my package with are getting properly templated for both manifests and values files before I deploy so I run `zarf package show-manifests zarf-package-podinfo-amd64.tar.zst -set=MY_VAR=my-val` and `zarf package show-values-files zarf-package-podinfo-amd64.tar.zst --set=MY_VAR=my-val`
```
Comment (soltysh) [inline 8-show-manifest-and-values/README.md:200]: I don't see how this story is different from the first one, other than the first is using directory, and this a zst file.

File: 8-show-manifest-and-values/README.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+Follow the [documentation style guide] for this section and the rest of the ZEP.
+Keep line lengths reasonable to make it easier for reviewers to provide
+feedback and reduce unnecessary changes.
+
+[documentation style guide]: https://docs.zarf.dev/contribute/style-guide/
+-->
+
+Both creators and deployers need a way to view their manifests and values files after Zarf variables are applied.
+
+This will be accomplished through new CLI commands `zarf dev show-manifests`, `zarf dev show-values-files`, `zarf package show-manifests`, and `zarf package show-values-files`
```
[the remark this answers is not in the document]
Comment (AustinAbro321) [reply 8-show-manifest-and-values/README.md:102]: That's actually how I had it originally, but I ended up changing it as it felt like two different commands. Originally I had written out `zarf package preview --show-manifests` and `zarf package preview --show-values-files`. However we need to be able to show the manifests and values file for already built packages as well as on packages before they're built so we needed to add a `zarf package inspect --show-manifests` and `zarf package inspect --show-values-file` command as well, zarf package inspect is already an existing command.  

I will build out the alternatives more with these suggestions so we can reevaluate. There's a few different ways we can do it. I think sub commands will likely be the way to go. IMO the biggest challenge here is that things need to work with both packages definitions and already built packages and certain flags will only be applicable to one or the other.

File: 8-show-manifest-and-values/README.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+this ZEP.  Describe why the change is important and the benefits to users. You
+can also optionally include links to [experience reports], [community slacks],
+or other references to show the community's interest in the ZEP.
+
+[experience reports]: https://go.dev/wiki/ExperienceReports
+[openssf slack]: https://openssf.slack.com/archives/C07AKUMBDMJ
+[kubernetes slack]: https://kubernetes.slack.com/archives/C03B6BJAUJ3
+-->
+
+Viewing manifests and values files after Zarf variable templating would be useful for both creators and deployers. Catching a mistake in templating early can reduce cycle time. A Helm template is almost instant, whereas create + deploy could take several minutes to hours.
```
[the remark this answers is not in the document]
Comment (AustinAbro321) [reply 8-show-manifest-and-values/README.md:117]: Yeah we have a `zarf dev lint` command. I can't find the issue for it, but I think there's some enhancements we could add to lint that would help here. For example, we could make sure that any variable defined is used somewhere in the package / manifests. And now that I think about it we could probably also make sure that after templating everything is valid yaml. 

Still I think users would want a way to check what their manifests are, and if variables are being set in the correct way.

File: 8-show-manifest-and-values/README.md (hunk 1 above)
Comment (AustinAbro321) [reply 8-show-manifest-and-values/README.md:169]: I think it's reasonable given that it would only prompt if there are variables that have the `prompt: true` key or templates that are not set (we error when templates aren't set)

File: 8-show-manifest-and-values/README.md (hunk 3 above)
Comment (AustinAbro321) [reply 8-show-manifest-and-values/README.md]: Good recs! I took them, but didn't use the commit feature because it messes with DCO :disappointed:

File: 8-show-manifest-and-values/README.md (hunk 4 above)
Comment (AustinAbro321) [reply 8-show-manifest-and-values/README.md]: I wanted to highlight that this will be used both before a package is created, and after a package is created. I added the different flags that the different commands have to make the stories more clear.

File: 0008-rework-inspect/README.md [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+or other references to show the community's interest in the ZEP.
+
+[experience reports]: https://go.dev/wiki/ExperienceReports
+[openssf slack]: https://openssf.slack.com/archives/C07AKUMBDMJ
+[kubernetes slack]: https://kubernetes.slack.com/archives/C03B6BJAUJ3
+-->
+
+Users need an easier way to view their package definition after it's rendered by Zarf, but before `zarf package create`. A rendered package definition has had templating, imports, and flavors applied. The only path to view the rendered package definition is running `zarf package create` and viewing the printed yaml before the (y/n) prompt. Having a separate command, `zarf dev inspect definition`, improves the UX by providing users with an easier way to view the rendered package definition. It also opens the possibility of allowing `zarf package create` to proceed without requiring user confirmation.
+
+Viewing manifests and values files after Zarf variable templating would be useful for both creators and deployers. Catching a mistake in templating early can reduce cycle time. A Helm template is almost instant, whereas create + deploy could take several minutes to hours.
```
Comment (soltysh) [inline 0008-rework-inspect/README.md:118]:
```suggestion
Viewing manifests and values files after Zarf variable templating would be useful for both creators and deployers. Catching a mistake in templating early can save time. A Helm template is almost instant, whereas create + deploy could take several minutes to hours.
```

File: 0008-rework-inspect/README.md [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+
+### Goals
+
+<!--
+List the specific goals of the ZEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- View manifests or values files after Zarf variable templating and Helm templating have been applied.
+- View rendered package definition before create
```
Comment (soltysh) [inline 0008-rework-inspect/README.md:135]:
```suggestion
- View rendered package definition before creation 
```

File: 0008-rework-inspect/README.md [hunk 9]
```diff
@@ -0,0 +1,414 @@
...
+for reviewers to clearly understand what you're proposing, but avoid including
+too many specifics like API designs or implementation details. Focus on the
+desired outcome and how success will be measured. The "Design Details" section
+below is for the real nitty-gritty.
+-->
+
+These commands will not have a confirm flag and will not prompt for optional components, package templates, or package variables. Users will be able to specify these values using flags, when applicable. None of these commands will run any Zarf actions.
+
+### zarf package inspect
+`zarf package inspect` will be deprecated and replaced by the five commands below
```
Comment (soltysh) [inline 0008-rework-inspect/README.md:161]:
```suggestion
`zarf package inspect` will be deprecated and replaced by the five commands below.
```

File: 0008-rework-inspect/README.md [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+#### zarf package inspect definition
+This will mirror behavior of the current `zarf package inspect <package>` command
```
Comment (soltysh) [inline 0008-rework-inspect/README.md:163]: Nit, I'd avoid printing help or limit code fragments in enhancements. Only where strictly necessary (eg. API types) it seems reasonable. All other instances are better expressed in plain English. This ensures that implementation and proposal don't drift.

**Review by soltysh — COMMENTED** [review]
A few minor nits, with the biggest one being the summary.

File: 0008-rework-inspect/README.md [hunk 11]
```diff
@@ -0,0 +1,414 @@
...
+A good summary should be at least a paragraph long.
+
+Follow the [documentation style guide] for this section and the rest of the ZEP.
+Keep line lengths reasonable to make it easier for reviewers to provide
+feedback and reduce unnecessary changes.
+
+[documentation style guide]: https://docs.zarf.dev/contribute/style-guide/
+-->
+
+The `zarf package inspect` command will become a parent command. It will have the five subcommands: `definition`, `sbom`, `images`, `manifests`, and `values-files`. The `package inspect` commands will accept an already built package, either local or OCI. `definition` and `images` will also accept a package in the cluster. The `zarf dev inspect` parent command will be introduced. It will have three subcommands: `definition`, `sbom` and `values-files`. These commands will accept a directory containing a `zarf.yaml` file.
```
[the remark this answers is not in the document]
Comment (AustinAbro321) [reply 0008-rework-inspect/README.md:101]: Gotcha, yeah summary should be aimed at the clear value end users are receiving from this change

File: 0008-rework-inspect/README.md (hunk 10 above)
Comment (AustinAbro321) [reply 0008-rework-inspect/README.md:163]: Fair point, there would definitely be drift. I will edit to remove code blocks

**Review by soltysh — APPROVED** [review]
lgtm
